Subject: [PATCH] layout: Start the session on the desktop

Use the desktop startup animation while keeping the overview available.

diff --git a/js/ui/layout.js b/js/ui/layout.js
--- a/js/ui/layout.js
+++ b/js/ui/layout.js
@@ -770,16 +770,14 @@
 
             const monitor = this.primaryMonitor;
 
-            if (!Main.sessionMode.hasOverview) {
-                const x = monitor.x + monitor.width / 2.0;
-                const y = monitor.y + monitor.height / 2.0;
+            const x = monitor.x + monitor.width / 2.0;
+            const y = monitor.y + monitor.height / 2.0;
 
-                this.uiGroup.set_pivot_point(
-                    x / global.screen_width,
-                    y / global.screen_height);
-                this.uiGroup.scale_x = this.uiGroup.scale_y = 0.75;
-                this.uiGroup.opacity = 0;
-            }
+            this.uiGroup.set_pivot_point(
+                x / global.screen_width,
+                y / global.screen_height);
+            this.uiGroup.scale_x = this.uiGroup.scale_y = 0.75;
+            this.uiGroup.opacity = 0;
 
             global.window_group.set_clip(monitor.x, monitor.y, monitor.width, monitor.height);
 
@@ -809,17 +807,13 @@
     }
 
     async _startupAnimationSession() {
-        if (Main.sessionMode.hasOverview) {
-            await Main.overview.runStartupAnimation();
-        } else {
-            await this.uiGroup.easeAsync({
-                scale_x: 1,
-                scale_y: 1,
-                opacity: 255,
-                duration: STARTUP_ANIMATION_TIME,
-                mode: Clutter.AnimationMode.EASE_OUT_QUAD,
-            }).catch(logErrorUnlessCancelled);
-        }
+        await this.uiGroup.easeAsync({
+            scale_x: 1,
+            scale_y: 1,
+            opacity: 255,
+            duration: STARTUP_ANIMATION_TIME,
+            mode: Clutter.AnimationMode.EASE_OUT_QUAD,
+        }).catch(logErrorUnlessCancelled);
     }
 
     _startupAnimationComplete() {
